### PR TITLE
Update(free-programming-books-zh.md): TypeScript

### PR DESCRIPTION
**Pull request by onlyling**
add TypeScript

## What does this PR do?
Add Resource(s)

### Description 

TypeScript - JavaScript that scales.

### For book lists, is it a book?

Not all books, some some are documents.

### Checklist:
- [x] Not a duplicate
- [ ] Included author(s) if appropriate
- [x] Lists are in alphabetical order
- [ ] Needed indications added (PDF, access notes, under construction)

### CODE REVIEW

**Comment by eshellman** [comment]
#2503 may help

File: free-programming-books-zh.md [hunk 1]
```diff
@@ -76,6 +76,7 @@
   * [Vim](#vim)
   * [Visual Prolog](#visual-prolog)
   * [WebAssembly](#webassembly)
+  * [TypeScript](#typescript)
```
Comment (bluetata) [inline free-programming-books-zh.md]: At first, you'd better to read the [CONTRIBUTING.md
](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)   

for [TypeScript] menu you should add it ref below image   

All the lists start with an Index. The idea is to list and link all sections and subsections there. Keep it in **alphabetical order.**

image

File: free-programming-books-zh.md [hunk 2]
```diff
@@ -740,3 +741,11 @@
...
 
 * [C/C++面向WebAssembly编程](https://github.com/3dgen/cppwasm-book)
+
+
+### TypeScript
+
+* [TypeScript 英文官网](https://www.typescriptlang.org)
+* [TypeScript 中文网](https://www.tslang.cn)
+* [TypeScript 入门教程](http://www.runoob.com/w3cnote/getting-started-with-typescript.html)
+* [TypeScript Deep Dive 中文版](https://github.com/jkchao/typescript-book-chinese)
```
Comment (bluetata) [inline free-programming-books-zh.md]: Move it to

image

**Comment by bluetata** [comment]
先阅读我上面写的 CONTRIBUTING.md 文件，每一个目录和电子书的链接都需要按照字母排序，并且每个section相互间都空两个空行

File: free-programming-books-zh.md [hunk 3]
```diff
@@ -720,6 +721,14 @@
 * [《The Swift Programming Language》中文版](https://www.gitbook.com/book/numbbbbb/-the-swift-programming-language-/details)
 
 
+### TypeScript
+
+* [TypeScript 英文官网](https://www.typescriptlang.org)
+* [TypeScript 中文网](https://www.tslang.cn)
+* [TypeScript 入门教程](https://www.runoob.com/w3cnote/getting-started-with-typescript.html)
+* [TypeScript Deep Dive 中文版](https://github.com/jkchao/typescript-book-chinese)
+
```
Comment (bluetata) [inline free-programming-books-zh.md]: 现在的机器check message为:
> 726:1-729:80  warning  Alphabetical ordering: swap l.727 and l.726  alphabetize-lists  remark-lint
  726:1-729:80  warning  Alphabetical ordering: swap l.729 and l.728  alphabetize-lists  remark-lint

在726行到729行的4个有顺序错误，你可以调整这4本电子书的顺序来使机器check消除掉，729行的应该放在第一个，其他的三个你需要自己尝试下。

**Comment by bluetata** [comment]
关于 机器check的可以参考 #3084

File: free-programming-books-zh.md [hunk 4]
```diff
@@ -720,6 +721,14 @@
 * [《The Swift Programming Language》中文版](https://www.gitbook.com/book/numbbbbb/-the-swift-programming-language-/details)
 
 
+### TypeScript
+
+* [TypeScript Deep Dive 中文版](https://github.com/jkchao/typescript-book-chinese)
+* [TypeScript 中文网](https://www.tslang.cn)
+* [TypeScript 入门教程](https://www.runoob.com/w3cnote/getting-started-with-typescript.html)
+* [TypeScript 英文官网](https://www.typescriptlang.org)
+
```
Comment (bluetata) [inline free-programming-books-zh.md]: <TypeScript 英文官网 >这个电子书建议你可以加在英文版本的md文件中，zh只针对中文的电子书

**Comment by eshellman** [comment]
@onlyling welcom to free-programming-books!

**Pull request merged**